spec/controllers/categories_controller_spec.rb

```diff
@@ -73,9 +78,90 @@
         expect(assigns(:products).count).to eq(4)
       end
 
+      it "only includes active / not retired products" do
+        product = Product.create(name: "blaglagolag", price: 1, seller_id: 1,
+          stock: 1, retired: true)
+        category.products << product
+
+        get :show, { id: category.id }
+        expect(assigns(:products)).not_to include(product)
+      end
+
       context "when a cateogry has no products" do
         # TODO: what do we want to have happen when there are no products?
       end
     end
+
+    describe "GET #new" do
+      it "responds successfully with an HTTP 200 status code" do
+        seller
+        session[:seller_id] = seller.id
+
+        get :new, seller_id: seller
+        expect(response).to be_success
+        expect(response).to have_http_status(200)
+      end
+
+      it "renders the new view" do
+        seller
+        session[:seller_id] = seller.id
+
+        get :new, seller_id: seller
+        expect(response).to render_template("new")
+      end
+
+      it "only lets a logged in seller create a category" do
+        seller
+
+        get :new
+        expect(response).to redirect_to(login_path)
+      end
+    end
+
+    describe "POST #create" do
+      context "valid category params" do
+        it "creates a category" do
+          seller
+          session[:seller_id] = seller.id
+
+          post :create, valid_params
```

Comment:
The first three lines of code in this test are exactly the same as the first three lines in all of the tests below. You should create a separate method that does this same operation each time rather than duplicating the test code every time.